File: problems_001to025/problem_019/solution.py
import os
import sys

PROJECT_PATH = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
sys.path.append(PROJECT_PATH)

from measurement import Stopwatch


class Solution(Stopwatch):
    def __init__(self):
        super().__init__()

    def execute(self):
        answer = 0
        day_of_week = 1
        for year in range(1900, 2001):
            for month in range(1, 13):
                if year >= 1901 and day_of_week == 0:
                    answer += 1
                if month == 4 or month == 6 or month == 9 or month == 11:
                    day_of_week += 30
                elif month != 2:
                    day_of_week += 31
                else:
                    if (year%4 == 0 and year%100 != 0) or year%400 == 0:
                        day_of_week += 29
                    else:
                        day_of_week += 28
                day_of_week %= 7
        return answer

if __name__ == '__main__':
    soln = Solution()
    result = soln.execute()
    print(result)

File: problems_001to025/problem_019/solution.js
const solution = () => {
  let answer = 0;

  let dayOfWeek = 1;

  for (let year = 1900; year <= 2000; year += 1) {
    for (let month = 1; month <= 12; month += 1) {
      if (year >= 1901 && dayOfWeek === 0) {
        answer += 1;
      }

      if ([4, 6, 9, 11].includes(month)) {
        dayOfWeek += 30;
      }
      else if (month !== 2) {
        dayOfWeek += 31;
      }
      else if ((year % 4 === 0 && year % 100 !== 0) || year % 400 === 0) {
        dayOfWeek += 29;
      }
      else {
        dayOfWeek += 28;
      }

      dayOfWeek %= 7;
    }
  }

  return answer;
};

if (process.env.NODE_ENV !== 'test') {
  // eslint-disable-next-line no-console
  console.log(solution());
}

module.exports = solution;
